### DISCUSSION TITLE
bug about Audio functions

### DISCUSSION THREAD

**Discussion by ghost** [edrlab/thorium-reader#2482] [General]
I use this program as main tool for language learning with epub3 files. (excellent work :) So, I occasionally stop audio for copying text into clipboard, pause audio and seek audio next to next line or previous line. I encountered numerous of bugs.
- When AudioStop activated, it seems AudioNext and AudioPrevious button work randomly seeking to anywhere they want.
- AudioNext and AudioPrevious activate autoplay even I pause the audio.
- when I swap windows with alt+tab sometimes the audio position is seeking randomly.
- I need to carefully click on thorium free space on the screen after using other programs to avoid accidentally playing unwanted audio lines.
- left click, right click and text selection appear the same behavior, play the audio line. I can't copy text while the audio is running.

**Comment by panaC** [comment]
Hello, can you please check with the latest thorium version 3.0.

**Comment by danielweck** [comment]
> When AudioStop activated, it seems AudioNext and AudioPrevious button work randomly seeking to anywhere they want.

This sounds like a bug. "stop" means stop, so "next" and "previous" shouldn't activate playback until "play" is activated. We need to check that the keyboard shortcuts don't cause user experience inconsistencies. As for the "random" nature of the playback position, Thorium resumes from the currently-known reading / listening position.

**Comment by danielweck** [comment]
> AudioNext and AudioPrevious activate autoplay even I pause the audio.

that's "normal" in the sense that the feature is intentionally implemented this way. the "paused" state resumes when the user interacts with one of the commands, or by clicking the text.

> left click, right click and text selection appear the same behavior, play the audio line. I can't copy text while the audio is running.

...that's the "click" interaction causing the "paused" state to resume.

**Comment by danielweck** [comment]
> when I swap windows with alt+tab sometimes the audio position is seeking randomly.

this is annoying. I wonder why?

**Comment by danielweck** [comment]
> I need to carefully click on thorium free space on the screen after using other programs to avoid accidentally playing unwanted audio lines.

I am not sure what you mean exactly, but my guess is that overall the "paused" state is too easy to reactivate into "resumed" audio.
we need to think about how to improve the UX.
